types/part_set.go

```diff
@@ -155,6 +155,9 @@ type PartSet struct {
 	parts         []*Part
 	partsBitArray *bits.BitArray
 	count         uint32
+	// a count of the total size (in bytes). Used to ensure that the
+	// part set doesn't exceed the maximum block bytes
+	byteSize int64
```

Comment:
It would've been even simpler to just calculate the size on the fly in `ByteSize()`, this isn't performance critical.

Reply:
Yeah true but it's a relatively minor difference right?

Reply:
Just less stuff that can go wrong :)